addons/block_code/ui/blocks/utilities/parameter_input/parameter_input.gd

```diff
@@ -125,13 +133,21 @@ func get_string() -> String:
 
 	match variant_type:
 		TYPE_STRING:
+			# HACK: don't include quotes around NIL strings
+			if input.ends_with("__nil__"):
+				return input.trim_suffix("__nil__")
 			return "'%s'" % input.replace("\\", "\\\\").replace("'", "\\'")
 		TYPE_VECTOR2:
-			return "Vector2(%s)" % input
+			return "Vector2%s" % str(input)
 		TYPE_COLOR:
 			return "Color%s" % str(input)
+		TYPE_OBJECT:
+			if input is OptionData:
+				var option_data := input as OptionData
+				return option_data.items[option_data.selected]
 		_:
 			return "%s" % input
+	return ""
```

Comment:
I think this is unnecessary since the `_` case will catch everything.

Reply:
I thought the same when looking at @wnbaum branch. But I remove this line, the syntax parser complains for the function saying "Not all code paths return a value.". So I understand why he added it.